pages/task/list/index.js

```diff
@@ -1,9 +1,33 @@
 const { Router } = require('express');
+const bodyParser = require('body-parser');
 
-module.exports = (settings = {}) => {
+module.exports = () => {
   const app = Router();
 
   app.get('/', (req, res) => res.redirect(req.buildRoute('dashboard')));
 
+  app.put('/assign', bodyParser.json(), (req, res, next) => {
+    let { assignedTo, taskId } = req.body;
+
+    if (!assignedTo || !taskId) {
+      return next();
+    }
+
+    const params = {
+      method: 'PUT',
+      json: {
+        data: {
+          profileId: assignedTo
+        }
+      }
+    };
+
+    req.api(`/tasks/${taskId}/assign`, params)
+      .then(response => {
+        res.json(response.json);
+      })
+      .catch(next);
+  });
```

Comment:
Is there a strong reason to add a new endpoint rather than using the existing assign endpoint?

Reply:
The current route is mounted under task.read and handles a form POST + redirect back to the task read page:
https://github.com/UKHomeOffice/asl-pages/blob/master/pages/task/read/routers/assign.js#L17-L43

We could replace the existing one with this background request one if you like?

Reply:
Yeah I realised as soon as I hit send, it's a bit nasty to have two assignment routes, but not a big deal really.